### PR TITLE
Add revision 1 and revision 2 to ProjectKB Alice PCB

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Adding a new revision of the ProjectKB Alice PCB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by noroadsleft** [comment]
> So rules.mk can go either way, either
> 
>     1. `keyboards/projectkb/alice/rules.mk` has all the common stuff, feature enabled/disabled logic and the rev `rules.mk` contains just overrides
> 
>     2. `keyboards/projectkb/alice/rules.mk` is kinda empty (in this case just the default folder) and the rev `rules.mk` contains everything
> 
> Would be happy with either, maybe a slight preference for 2.

I prefer Option 2 because eventually I hope we can completely do away with `DEFAULT_FOLDER`.

File: keyboards/projectkb/alice/rules.mk [hunk 1]
```diff
@@ -1,22 +1,2 @@
-# MCU name
-MCU = STM32F072
+DEFAULT_FOLDER = projectkb/alice/rev2
```
Comment (noroadsleft) [inline keyboards/projectkb/alice/rules.mk:1]: IMO this should point at rev1 to maintain existing behaviour for existing users, with the goal of eventually removing this file entirely.

```suggestion
DEFAULT_FOLDER = projectkb/alice/rev1
```

File: keyboards/projectkb/alice/rev1/rules.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+WS2812_DRIVER = spi
```
Comment (noroadsleft) [inline keyboards/projectkb/alice/rev1/rules.mk:18]: Alignment and templating changes. No logic change.

```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
WS2812_DRIVER = spi
```

File: keyboards/projectkb/alice/rev2/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+WS2812_DRIVER = spi
```
Comment (noroadsleft) [inline keyboards/projectkb/alice/rev2/rules.mk:18]: Same as above.

```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
WS2812_DRIVER = spi
```

**Comment by awkannan** [comment]
@zvecr @noroadsleft thanks again for the feedback! Updated.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**